server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -82,18 +82,14 @@
     private MutableQueryTimeout cancellable;
 
     public ContextIndexSearcher(IndexReader reader, Similarity similarity,
-                                QueryCache queryCache, QueryCachingPolicy queryCachingPolicy) throws IOException {
-        this(reader, similarity, queryCache, queryCachingPolicy, new MutableQueryTimeout());
+                                QueryCache queryCache, QueryCachingPolicy queryCachingPolicy,
+                                boolean wrapWithExitableDirReader) throws IOException {
+        this(reader, similarity, queryCache, queryCachingPolicy, wrapWithExitableDirReader ? new MutableQueryTimeout() : null);
```

Comment:
We use the cancellable  even if the low level cancellation is set to false (query timeout option and bulk scorer) so we should never set the cancellable to `null` ?

Reply:
Yep, sure, I just pushed it to check some other compilation stuff. it's not ready yet, but thx!